File: python-client-sdk/src/orbapi/models/operations/get_subscriptions_by_coupon_id.py
from __future__ import annotations
import dataclasses
from ..shared import subscription as shared_subscription
from dataclasses_json import Undefined, dataclass_json
from orbapi import utils
from typing import Any, Optional


@dataclasses.dataclass
class GetSubscriptionsByCouponIDPathParams:
    coupon_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'coupon_id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetSubscriptionsByCouponIDRequest:
    path_params: GetSubscriptionsByCouponIDPathParams = dataclasses.field()
    

@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class GetSubscriptionsByCouponID200ApplicationJSON:
    data: Optional[list[shared_subscription.Subscription]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data'), 'exclude': lambda f: f is None }})
    pagination_metadata: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pagination_metadata'), 'exclude': lambda f: f is None }})
    

@dataclasses.dataclass
class GetSubscriptionsByCouponIDResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    get_subscriptions_by_coupon_id_200_application_json_object: Optional[GetSubscriptionsByCouponID200ApplicationJSON] = dataclasses.field(default=None)
    

File: typescript-client-sdk/src/sdk/models/operations/getsubscriptionsbycouponid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Expose, Type } from "class-transformer";


export class GetSubscriptionsByCouponIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=coupon_id" })
  couponId: string;
}

export class GetSubscriptionsByCouponIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetSubscriptionsByCouponIdPathParams;
}

export class GetSubscriptionsByCouponId200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: shared.Subscription })
  @Expose({ name: "data" })
  @Type(() => shared.Subscription)
  data?: shared.Subscription[];

  @SpeakeasyMetadata()
  @Expose({ name: "pagination_metadata" })
  paginationMetadata?: Record<string, any>;
}

export class GetSubscriptionsByCouponIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getSubscriptionsByCouponId200ApplicationJSONObject?: GetSubscriptionsByCouponId200ApplicationJSON;
}